pkg/gatherers/conditional/gather_containers_logs.go

```diff
@@ -33,52 +34,67 @@ func (g *Gatherer) BuildGatherLogsOfUnhealthyPods(paramsInterface interface{}) (
 			if err != nil {
 				return nil, []error{err}
 			}
-			return g.gatherLogsOfUnhealthyPods(ctx, kubeClient.CoreV1(), params)
+			coreClient := kubeClient.CoreV1()
+			return g.gatherContainersLogs(ctx, params, coreClient)
 		},
 		CanFail: canConditionalGathererFail,
 	}, nil
 }
 
-func (g *Gatherer) gatherLogsOfUnhealthyPods(
-	ctx context.Context, coreClient v1.CoreV1Interface, params GatherLogsOfUnhealthyPodsParams,
+func (g *Gatherer) gatherContainersLogs(
+	ctx context.Context,
+	params GatherContainersLogsParams,
+	coreClient corev1client.CoreV1Interface,
 ) ([]record.Record, []error) {
-	errs := []error{}
-	records := []record.Record{}
-
 	alertInstances, ok := g.firingAlerts[params.AlertName]
 	if !ok {
-		return nil, []error{fmt.Errorf("conditional gatherer triggered, but specified alert %q is not firing", params.AlertName)}
+		err := fmt.Errorf("condigional gather triggered, but specified alert %q is not firing", params.AlertName)
+		return nil, []error{err}
 	}
+
+	var errs []error
+	var records []record.Record
+
 	for _, alertLabels := range alertInstances {
-		alertNamespace, err := getAlertPodNamespace(alertLabels)
+		alertPodNamespace, err := getAlertPodNamespace(alertLabels)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
-
-		alertPod, err := getAlertPodName(alertLabels)
+		alertPodName, err := getAlertPodName(alertLabels)
 		if err != nil {
 			errs = append(errs, err)
 			continue
 		}
+		var alertPodContainer string
+		if len(params.Container) > 0 {
+			alertPodContainer = params.Container
+		} else {
+			alertPodContainer, err = getAlertPodContainer(alertLabels)
+			if err != nil {
+				errs = append(errs, err)
+			}
+		}
+
+		logContainersFilter := common.LogContainersFilter{
+			Namespace:     alertPodNamespace,
+			FieldSelector: fmt.Sprintf("metadata.name=%s", alertPodName),
+		}
 
 		// The container label may not be present for all alerts (e.g., KubePodNotReady).
-		containerFilter := ""
-		if alertContainer, ok := alertLabels["container"]; ok && alertContainer != "" {
-			containerFilter = fmt.Sprintf("^%s$", alertContainer)
+		if len(alertPodContainer) > 0 {
+			logContainersFilter.ContainerNameRegexFilter = fmt.Sprintf("^%s$", alertPodContainer)
```

Comment:
Do we have to do this when the container name (`params.Container`) is given? This looks like we will always be searching the containers by the regex, not sure if it's necessary. 
I would also call the variable just `containerName` instead of `alertPodContainer`....just my 2 cents.

Reply:
OK it's probably not a big deal, when I am looking at it again, but maybe we could think of updating the `CollectLogsFromContainers` to have some attribute/param for the container name.

Reply:
I have no strong opinions about it. We could probably handle it in some other task.